plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/app/DefaultCertificateStorage.java

```diff
@@ -138,14 +135,24 @@ public void addCertificate(@NotNull DBPDataSourceContainer dataSource, @NotNull
             CertificateFactory cf = CertificateFactory.getInstance("X.509");
             List<Certificate> certChain = new ArrayList<>();
             if (caCertData != null) {
-                Certificate caCert = cf.generateCertificate(new ByteArrayInputStream(caCertData));
-                keyStore.setCertificateEntry(CA_CERT_ALIAS, caCert);
+                Collection<? extends Certificate> certificates = cf.generateCertificates(new ByteArrayInputStream(
+                    caCertData));
+                int i = 0;
+                for (Certificate certificate : certificates) {
+                    keyStore.setCertificateEntry(i == 0 ? CA_CERT_ALIAS : CA_CERT_ALIAS + i, certificate);
+                    i++;
+                }
```

Comment:
Ugh.. how about a traditional for-loop if you need a counter?

Reply:
Collection<?> requires additional casts to use counter